packages/@sanity/desk-tool/src/components/confirmDeleteDialog/ConfirmDeleteDialogBody.tsx

```diff
@@ -0,0 +1,198 @@
+  CrossDatasetReferences,
+  CrossDatasetReferencesSummary,
+  Chevron,
+  TableContainer,
+  Table,
+} from './ConfirmDeleteDialogBody.styles'
+
+type DeletionConfirmationDialogBodyProps = Required<ReferringDocuments> & {
+  documentTitle: React.ReactNode
+  action: string
+}
+
+function getSchemaType(typeName: string): SchemaType | null {
+  const schemaMod = require('part:@sanity/base/schema')
+  const schema = schemaMod.default || schemaMod
+  return schema.get(typeName) || null
+}
+
+/**
+ * The inner part of the `ConfirmDeleteDialog`. This is ran when both the
+ * `crossDatasetReferences` and `internalReferences` are loaded.
+ */
+export function ConfirmDeleteDialogBody({
+  crossDatasetReferences,
+  internalReferences,
+  documentTitle,
+  total,
+  action,
+}: DeletionConfirmationDialogBodyProps) {
+  const toast = useToast()
+
+  if (internalReferences?.total === 0 && crossDatasetReferences?.total === 0) {
+    return (
+      <Text as="p">
+        Are you sure you want to delete <strong>“{documentTitle}”</strong>?
+      </Text>
+    )
+  }
+
+  return (
+    <>
+      <Card padding={3} radius={2} tone="caution">
+        <Flex>
+          <Text size={1}>
+            <WarningOutlineIcon />
+          </Text>
+          <Box flex={1} marginLeft={3}>
+            <Text size={1}>
+              {total === 1 ? (
+                <>1 document refers to “{documentTitle}”</>
+              ) : (
+                <>
+                  {total.toLocaleString()} documents refer to “{documentTitle}”
+                </>
+              )}
+            </Text>
+          </Box>
+        </Flex>
+      </Card>
+
+      <Text>
+        You may not be able to delete “{documentTitle}” because the following documents refer to it:
+      </Text>
+
+      <ReferencesCard radius={2} shadow={1}>
+        {internalReferences.total > 0 && (
+          <InternalReferences>
+            {internalReferences.references.map((internalReference) => {
+              const type = getSchemaType(internalReference._type)
+              // TODO: this handle this case better
+              if (!type) return null
+
+              return (
+                <Box as="li" key={internalReference._id} paddingX={3} paddingY={3}>
+                  <SanityPreview type={type} value={internalReference} layout="default" />
+                </Box>
+              )
+            })}
+            {internalReferences.total > internalReferences.references.length && (
+              <Box as="li" padding={3}>
+                <OtherReferenceCount {...internalReferences} />
+              </Box>
+            )}
+          </InternalReferences>
+        )}
+
+        {crossDatasetReferences.total > 0 && (
+          <CrossDatasetReferences
+            style={{
+              // only add the border if needed
+              borderTop:
+                internalReferences.total > 0
+                  ? '1px solid var(--card-shadow-outline-color)'
+                  : undefined,
+            }}
```

Comment:
This could also be solved with a transient prop:

```tsx
const CrossDatasetReferences = styled.details`
  border-top: ${props => props.$borderTop && '1px solid var(--card-shadow-outline-color)'};
  /* ... */
`

<CrossDatasetReferences $borderTop={internalReferences.total > 0} />
```